pkg/parser/paragraph_test.go

```diff
@@ -167,6 +167,22 @@ foo`
 				}
 				Expect(ParseDraftDocument(source)).To(MatchDraftDocument(expected))
 			})
+
+			It("with counters", func() {
+				source := `foo{counter:foo} bar{counter2:foo} baz{counter:foo} bob{counter:bob}`
```

Comment:
the Asciidoctor docs says that `counter2` can be used to increment the counter without displaying it: https://asciidoctor.org/docs/user-manual/#counters

Also, when processing the source above, Asciidoctor produces the following result:
```
<div class="paragraph">
<p>foo1 bar baz3 bob1</p>
</div>
```

Reply:
Whoa.  I  totally misread the counter2 docs -- I though it was display without incrementing, not the reverse.  Will fix.

Reply:
I will come back with a revised PR with that fixed.

I recommend that we integrate the fix for the stray debugging message (you can cherry-pick or I can submit a separate PR for that), and then I will squash this all together along with the fixes.  (It is not convenient to keep them separate, as I have to refactor the CounterSubstitution a bit to fix the revised understanding of counter2.)